File: id/src/main/java/com/simprints/id/data/db/remote/enums/REFUSAL_FORM_REASON.java
package com.simprints.id.data.db.remote.enums;

@Deprecated
public enum REFUSAL_FORM_REASON {
    SCANNER_NOT_WORKING("scanner_not_working"),
    REFUSED("refused"),
    OTHER("other");

    private String title;

    REFUSAL_FORM_REASON(String title) {
        this.title = title;
    }

    @Override
    public String toString() {
        return title;
    }
}

File: fingerprint/controller/src/main/java/com/simprints/fingerprint/data/domain/refusal/RefusalFormReason.kt
package com.simprints.fingerprint.data.domain.refusal

import com.simprints.feature.exitform.config.ExitFormOption

/**
 * This enum class represents the different reasons for submitting a refusal form.
 */

enum class RefusalFormReason {
    REFUSED_RELIGION,
    REFUSED_DATA_CONCERNS,
    REFUSED_PERMISSION,
    SCANNER_NOT_WORKING,
    REFUSED_NOT_PRESENT,
    REFUSED_YOUNG,
    APP_NOT_WORKING,
    OTHER;

    companion object {
        fun fromExitFormOption(option: ExitFormOption) = when (option) {
            ExitFormOption.ReligiousConcerns -> REFUSED_RELIGION
            ExitFormOption.DataConcerns -> REFUSED_DATA_CONCERNS
            ExitFormOption.NoPermission -> REFUSED_PERMISSION
            ExitFormOption.AppNotWorking -> APP_NOT_WORKING
            ExitFormOption.ScannerNotWorking -> SCANNER_NOT_WORKING
            ExitFormOption.PersonNotPresent -> REFUSED_NOT_PRESENT
            ExitFormOption.TooYoung -> REFUSED_YOUNG
            ExitFormOption.Other -> OTHER
        }
    }
}
